addons/mail/models/discuss/discuss_channel.py

```diff
@@ -271,33 +271,25 @@ def write(self, vals):
             failing_channels = self.filtered(lambda channel: channel.channel_type != vals.get('channel_type'))
             if failing_channels:
                 raise UserError(_('Cannot change the channel type of: %(channel_names)s', channel_names=', '.join(failing_channels.mapped('name'))))
+        old_vals = {channel.id: channel._channel_basic_info() for channel in self}
+        result = super().write(vals)
         notifications = []
         for channel in self:
-            current_val = channel.read(vals.keys())[0]
+            info = channel._channel_basic_info()
             diff = {}
-            for key in vals.keys():
-                if current_val.get(key) != vals.get(key) and key != "image_128":
-                    diff[key] = vals[key]
+            for key, value in info.items():
+                if value != old_vals.get(channel.id)[key]:
+                    diff[key] = value
             if diff:
                 notifications.append([channel, "mail.record/insert", {
                     "Thread": {
-                        "id": current_val["id"],
-                        "model": "discuss.channel",
+                        "id": info["id"],
```

Comment:
not mandatory needed, but this makes more sense to me.

Reply:
Why do you say it's not needed? The id is important to know which record is updated.

Reply:
It should just be `channel.id` I think, that's the most readable.